### PR TITLE
[H] Support setting the elevation of disabled floating action buttons

### PR DESCRIPTION
**Pull request by Hixie**
Previously, a disabled floating action button always had zero
elevation, which looks dumb.

This also fixes the issue whereby highlightElevation was not honoured
on floating action buttons.

This also fixes an issue I found during testing whereby setState was
being called during build when onHighlightChanged fired due to
onPressed becoming null while a gesture is ongoing (which triggers an
onTapCancel synchronously during build).

### CODE REVIEW

**Comment by Hixie** [comment]
cc @HansMuller 

This will also require a golden file update but I'll wait until the tree is open and I'm ready to land before rolling the goldens repo.

File: packages/flutter/lib/src/material/floating_action_button.dart [hunk 1]
```diff
@@ -43,23 +43,29 @@ class _DefaultHeroTag {
 ///
 /// Use at most a single floating action button per screen. Floating action
 /// buttons should be used for positive actions such as "create", "share", or
-/// "navigate".
+/// "navigate". (If more than one floating action buttos is used within a
```
Comment (GaryQian) [inline packages/flutter/lib/src/material/floating_action_button.dart]: Did you mean `button`?

**Comment by dnfield** [comment]
(I reverted the changes for this in goldens - they'll have to be added back when this is ready)

File: packages/flutter/lib/src/material/floating_action_button.dart (hunk 1 above)
Comment (Hixie) [reply packages/flutter/lib/src/material/floating_action_button.dart]: fixed

**Comment by zoechi** [comment]
Related to #227

**Comment by Hixie** [comment]
This is ready and awaiting review.

**Review by HansMuller — APPROVED** [review]
Sorry about the monumental review delay.
LGTM

**Pull request merged**